src/storage-client/src/source/persist_source.rs

```diff
@@ -250,11 +250,14 @@ where
             return;
         }
 
-        let read = persist_clients_stream
-            .lock()
-            .await
-            .open(persist_location_stream)
-            .await;
+        let read = {
+            let mut persist_clients = persist_clients_stream.lock().await;
+
+            let read = persist_clients.open(persist_location_stream).await;
+            std::mem::drop(persist_clients);
+
+            read
+        };
```

Comment:
```suggestion
        let read = {
            let mut persist_clients = persist_clients_stream.lock().await;
            persist_clients.open(persist_location_stream).await;
        };
```

I think is more clear! but this is a nit